File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem08_IncreasingTripletSubsequence.java
package org.phoenix.leetcode.challenges;

public class Problem08_IncreasingTripletSubsequence {
    public boolean increasingTriplet(int[] nums) {
        int min = Integer.MAX_VALUE;
        int secMin = Integer.MAX_VALUE;
        for (int num : nums) {
            if (num <= min) min = num;
            else if (num <= secMin) secMin = num;
            else return true;
        }
        return false;
    }
}

File: JavaScript/leetcode75/src/main/Problem08_IncreasingTripletSubsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    let min = Number.MAX_VALUE;
    let secMin = Number.MAX_VALUE;
    for (let num of nums) {
        if (num <= min) min = num;
        else if (num <= secMin) secMin = num;
        else return true;
    }
    return false;
};

module.exports = increasingTriplet;